lock-api/src/main/java/com/palantir/lock/watch/LockWatchEventCache.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+public interface LockWatchEventCache {
```

Comment:
Again, this is a dodgy diff - should not really have changed much, but moving it seemed to have broken github's diffs

Also, should not be different from interface in https://github.com/palantir/atlasdb/pull/4811 (any differences will be resolved when they merge).